hypervisor/common/hypercall.c

```diff
@@ -676,7 +676,7 @@ int32_t hcall_set_vm_memory_regions(struct acrn_vm *vm, uint64_t param)
 	        ret = -EFAULT;
 	} else {
 		target_vm = get_vm_from_vmid(regions.vmid);
-		if ((target_vm == NULL) || is_sos_vm(target_vm)) {
+		if ((target_vm == NULL) || is_sos_vm(target_vm) || (target_vm->state == VM_STATE_INVALID)) {
```

Comment:
How about pre-launched VM?

Reply:
@dongyaozu  have revised this patch, prevent pre-launched VM from calling hcall_set_vm_memory_regions() hypercall. please review, thanks.